File: output/deepseek_chat/avatar/algo_based_translation/python/java/atcoder_AGC017_A.java

import java.util.*;

public class atcoder_AGC017_A {
    public static double combi(int N, int K) {
        double a = 1.0;
        for (int i = 0; i < K; i++) {
            a *= (N - i);
        }
        for (int j = 0; j < K; j++) {
            a /= (j + 1);
        }
        return a;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int P = scanner.nextInt();
        int ans = 0;
        scanner.nextLine();
        String[] inputList = scanner.nextLine().split(" ");
        List<Integer> lis = new ArrayList<>();
        for (String num : inputList) {
            lis.add(Integer.parseInt(num));
        }
        List<Integer> ls = new ArrayList<>();
        for (int a : lis) {
            ls.add(a % 2);
        }
        int one = Collections.frequency(ls, 1);
        int zero = Collections.frequency(ls, 0);
        double pattern_a = 0.0;
        double pattern_b = 0.0;
        for (int j = 0; j <= zero; j++) {
            pattern_b += combi(zero, j);
        }
        int time = 0;
        while (time <= one) {
            if (time % 2 == P) {
                pattern_a += combi(one, time);
            }
            time++;
        }
        System.out.println((int)(pattern_a * pattern_b));
    }
}

File: dataset/avatar/Python/Code/atcoder_AGC017_A.py
N , P = map ( int , input ( ).split ( ) )
def combi ( N , K ) :
    a = 1
    for i in range ( K ) :
        a *= N - i
    for j in range ( K ) :
        a /= j + 1
    return a
ans = 0
lis = list ( map ( int , input ( ).split ( ) ) )
ls = [ ]
for a in lis :
    ls.append ( a % 2 )
one = ls.count ( 1 )
zero = ls.count ( 0 )
pattern_a = 0
pattern_b = 0
for j in range ( zero + 1 ) :
    pattern_b += combi ( zero , j )
time = 0
while time <= one :
    if time % 2 == P :
        pattern_a += combi ( one , time )
    time += 1
print ( int ( pattern_a * pattern_b ) )

